File: solutions/julia/raindrops/1/raindrops.jl
function raindrops(number)
    sound = ""
    factors = Dict( 3 => "Pling", 5 => "Plang", 7 => "Plong" )
    for factor in sort(collect(keys(factors)))
        if number % factor == 0
            sound = string(sound, factors[factor])
        end
    end
    return sound == "" ? string(number) : sound
end

File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
    map <- list( list(3, "Pling"), list(5, "Plang"), list(7, "Plong") )
    sounds <- c()
    for ( n in c(1:length(map)) ) {
        factor <- map[[n]][[1]]
        sound  <- map[[n]][[2]]
        if ( number %% factor == 0 ) {
            sounds[ length(sounds) + 1 ] <- sound
         }
    }
    if ( length(sounds) == 0 ) {
        sounds <- number
    }

    paste( sounds, collapse='' )
}
